pkg/authorization/authorizer/adapter/attributes.go

```diff
@@ -70,6 +76,26 @@ func (a AdapterAttributes) GetNamespace() string {
 	return a.namespace
 }
 
+func (a AdapterAttributes) GetName() string {
+	return a.AuthorizationAttributes.GetResourceName()
+}
+
+func (a AdapterAttributes) GetSubresource() string {
+	tokens := strings.SplitN(a.AuthorizationAttributes.GetResource(), "/", 2)
+	if len(tokens) != 2 {
+		return ""
+	}
+	return tokens[1]
+}
+
+func (a AdapterAttributes) GetResource() string {
+	tokens := strings.SplitN(a.AuthorizationAttributes.GetResource(), "/", 2)
```

Comment:
Not now.  Eliminated it satisfying both interfaces.  Perfect round tripping
is still supported on the exact type.  It should never have been anonymous
and it should probably have never tried to satisfy both interfaces.

On Thu, Apr 14, 2016 at 2:19 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/authorization/authorizer/adapter/attributes.go
> <https://github.com/openshift/origin/pull/8484#discussion_r59765891>:
>
> > @@ -70,6 +76,26 @@ func (a AdapterAttributes) GetNamespace() string {
> >  	return a.namespace
> >  }
> >
> > +func (a AdapterAttributes) GetName() string {
> > +	return a.AuthorizationAttributes.GetResourceName()
> > +}
> > +
> > +func (a AdapterAttributes) GetSubresource() string {
> > +	tokens := strings.SplitN(a.AuthorizationAttributes.GetResource(), "/", 2)
> > +	if len(tokens) != 2 {
> > +		return ""
> > +	}
> > +	return tokens[1]
> > +}
> > +
> > +func (a AdapterAttributes) GetResource() string {
> > +	tokens := strings.SplitN(a.AuthorizationAttributes.GetResource(), "/", 2)
>
> this seems like it propagates the bug... AdapterAttributes satisfying
> oauthorizer.AuthorizationAttributes with a GetResource impl that does not
> include subresource is scary
>
> we should update oauthorizer.AuthorizationAttributes to also have a
> GetSubresource(), and update node_auth.go to set resource and subresource
> correctly
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8484/files/c627a3c0e36d1a4023f08e9e6ec3d0c4e62e3421..7e72550dea93a514bf6bfc30e087a46d6a23a3a7#r59765891>
>